pkg/stanza/fileconsumer/reader.go

```diff
@@ -102,33 +126,65 @@ func (r *Reader) Close() {
 			r.Debugw("Problem closing reader", zap.Error(err))
 		}
 	}
+
+	if r.header != nil {
+		if err := r.header.Shutdown(); err != nil {
+			r.Warnw("Problem shutting down header pipeline", zap.Error(err))
+		}
+	}
+}
+
+func min0(a, b int) int {
+	if a < 0 || b < 0 {
+		return 0
+	}
+	if a < b {
+		return a
+	}
+	return b
+}
+
+// fingerprintReader is an io.ReadSeeker that updates it's fingerprint as it reads.
+type fingerprintReader struct {
```

Comment:
Why do we need a new type?

Reply:
I moved this into it's own code because the offset it was looking at was owned by the reader, which made it difficult to update the fingerprint while reading in the header. 

This shared type has the actual file offset after it's seeked, so it can be shared between the header and reader.

Reply:
Removed this type with the rework to put header parsing in the ReadToEnd loop.